### DISCUSSION TITLE
Is there a CallDelayed(wait, action) method?

### DISCUSSION THREAD

**Discussion by kenofori** [Cysharp/UniTask#523] [Q&A] [answered]
Hi there,

I'm coming from MEC library, which has following method, any similar method in UniTask or do i have to go with creating a custom method that await ?

Thanks

Timing.CallDelayed(delayInSeconds, action);

**Comment by william-craven-pace** [comment, marked as answer]
```C#
UniTask.Delay(millisecondsDelay).ContinueWith(() => { });
// or
UniTask.Delay(TimeSpan.FromSeconds(delayInSeconds)).ContinueWith(() => { });
```

**Reply by kenofori** [reply]
Thank you very much @william-craven-pace 
How can i avoid garbage generation? should i pass a local method name or passing lambda is better

**Reply by timcassell** [reply]
Use async/await.

```cs
async UniTask Func()
{
    await UniTask.Delay(millisecondsDelay);
    ...
}
```

**Reply by william-craven-pace** [reply]
You can use a named method or lambda. You should be able to avoid the allocation by capturing the action and then assigning it or just do as @timcassell suggested.
